### PR TITLE
Fix issue 1479 - gluster-block error handling

### LINKED ISSUE #1479: heketi-cli is failing to provide an error message when gluster-block create commands fail
### Kind of issue
Bug

### Observed behavior

When a block volume is being created and the underlying `gluster-block create ...` command fails it returns json on stdout. Heketi is incorrectly looking only for output on stderr and return a "blank" error string up to the caller.

### Expected/desired behavior

A reasonable error string is best, but something is better than nothing.

### Details on how to reproduce (minimal and precise)

Prevent gluster-block create commands from returning successfully and run any `heketi-cli blockvolume create ....` command. No proper error message is returned.

### Information about the environment:

- Heketi version used (e.g. v6.0.0 or master): master

### PR DESCRIPTION
**Pull request by phlogistonjohn**
### What does this PR achieve? Why do we need it?

This PR updates the output handling in the code that calls gluster-block create so that the way that command reports errors is more correctly handled with the new command results model.

### Does this PR fix issues?

Fixes #1479

### CODE REVIEW

**Comment by obnoxxx** [comment]
```
Build timed out (after 180 minutes).
```
:-o

**Comment by obnoxxx** [comment]
@phlogistonjohn  Do we possibly need to do similar, more fine grained error handing for volume create as well? Or is it not needed, because gluster cli is better at doing stdout/stderr, etc?... at least we do not parse output of the gluster commands in `VolumeCreate()`.

**Comment by obnoxxx** [comment]
flaky pending ops test (5 vs 9...).

retest this please

**Comment by phlogistonjohn** [comment]
> @phlogistonjohn Do we possibly need to do similar, more fine grained error handing for volume create as well? Or is it not needed, because gluster cli is better at doing stdout/stderr, etc?... at least we do not parse output of the gluster commands in `VolumeCreate()`.

The "gluster" command certainly is a bit simpler and this was a regression brought in by the changes around the remoteexec stuff that bit the more complex parsing in the gluster-block stuff only.

I'd say we leave "gluster" stuff alone for now until we decide we need more complex handling for those commands.

**Comment by obnoxxx** [comment]
Failure like this:

```
18:50:10 --- FAIL: TestBlockVolumeOperation (47.81s)
18:50:10     assert.go:44: 
18:50:10         
ASSERT:	func (github.com/heketi/heketi/tests/functional/TestSmokeTest/tests.TestBlockVolumeOperation) 0x6ce644
18:50:10         
	File /root/go/src/github.com/heketi/heketi/tests/functional/TestSmokeTest/tests/block_volume_test.go:61
18:50:10         
	Info: [expected err == nil, got: Unable to parse block volume create results for blockvol_b2af490b593ebcd055c93b9745ef7e62]
18:50:10     assert.go:44: 
18:50:10         
ASSERT:	func (github.com/heketi/heketi/pkg/testutils.(*ClusterEnv).Teardown) 0x6c4693
18:50:10         
	File /root/go/src/github.com/heketi/heketi/pkg/testutils/cluster.go:197
18:50:10         
	Info: [expected err == nil, got: Cannot delete a block hosting volume containing block volumes]
```

THe others as well. It does not seem related to this patch...

**Comment by obnoxxx** [comment]
retest this please

**Comment by obnoxxx** [comment]
again the block volume operation error
what's going on???
https://ci.centos.org/job/gluster_heketi-functional/1867/consoleFull

**Comment by obnoxxx** [comment]
retest this please

**Comment by obnoxxx** [comment]
Now this error looks different:
```
21:06:30 --- FAIL: TestBlockVolumeOperation (83.31s)
21:06:30     assert.go:44: 
21:06:30         
ASSERT:	func (github.com/heketi/heketi/tests/functional/TestSmokeTest/tests.TestBlockVolumeOperation) 0x6ce644
21:06:30         
	File /root/go/src/github.com/heketi/heketi/tests/functional/TestSmokeTest/tests/block_volume_test.go:61
21:06:30         
	Info: [expected err == nil, got: Unable to parse block volume create results for blockvol_fbc4338efe8869510ef87d4a15101e42]
```
Might this be due to the changed error handling?

**Comment by obnoxxx** [comment]
ok. i had looked at too many PRs at the same time... ;-)
The error message is the same as above.
So this PR already brings improvement. the underlying error from gluster-block just looks different.

**Comment by phlogistonjohn** [comment]
Update with better handling of unparsable results

File: executors/cmdexec/block_volume.go [hunk 1]
```diff
@@ -52,19 +52,38 @@ func (s *CmdExecutor) BlockVolumeCreate(host string,
...
+	if output == "" {
+		output = results[0].ErrOutput
+	}
+
 	var blockVolumeCreate CliOutput
-	err = json.Unmarshal([]byte(results[0].Output), &blockVolumeCreate)
+	err = json.Unmarshal([]byte(output), &blockVolumeCreate)
 	if err != nil {
-		return nil, fmt.Errorf("Unable to get the block volume create info for block volume %v", volume.Name)
+		logger.Warning("Unable to gluster-block output [%v]: %v",
```
[the remark this answers is not in the document]
Comment (phlogistonjohn) [reply executors/cmdexec/block_volume.go]: Yep. Series updated.

**Pull request merged**